src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -1,11 +1,49 @@
 package core.basesyntax.service;
 
+import core.basesyntax.dao.StorageDaoImpl;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MIN_AGE = 18;
+    private static final int MAX_AGE = 100;
+    private static final int MIN_LENGTH = 6;
+    private final StorageDaoImpl storageDao;
+
+    public RegistrationServiceImpl() {
+        storageDao = new StorageDaoImpl();
+    }
 
     @Override
     public User register(User user) {
-        return null;
+        userExists(user);
+        validateLogin(user);
+        validateAge(user);
+        validatePassword(user);
+        return storageDao.get(user.getLogin()) == null ? storageDao.add(user) : null;
+    }
+
+    public void userExists(User user) {
+        if (user == null) {
+            throw new RuntimeException("There is no user");
+        }
+    }
+
+    public void validateLogin(User user) {
```

Comment:
better to use boolean instead of void, they are easier to test

Reply:
these methods will throw RuntimeException, otherwise program flow will continue, there is no sense for them to return any value. if there would be some checkable exception, it makes sense, we must know that we got problem here, but it's RuntimeException, program will be terminated.